code/datums/components/storage/storage.dm

```diff
@@ -420,6 +441,9 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
 	M.client.screen |= closer
 	M.client.screen |= real_location.contents
 	M.set_active_storage(src)
+	if(ismovable(real_location))
+		var/atom/movable/movable_loc = real_location
+		movable_loc.become_active_storage(src)
```

Comment:
fun fact: if you open your backpack and move, your mob then performs a BFS to find out whether its currently active storage object is still within reach. now i added active storage objects to important_recursive_contents so that if said object is within your inventory you early out of the BFS performed by CanReach()